src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ActiveDirectory.tests.ps1

```diff
@@ -11,51 +11,66 @@ InModuleScope Arcus.Scripting.ActiveDirectory {
 
-                # Act
-                {
-                   List-AzADAppRoleAssignments -ClientId $ClientId -RolesAssignedToClientId $RolesAssignedToClientId
-                } | Should -Not -Throw
+                try {
+                    # Act
+                    {
+                       List-AzADAppRoleAssignments -ClientId $MainAppClientId -RolesAssignedToClientId $ClientAppClientId
+                    } | Should -Not -Throw
+                } finally {
+                    Remove-AzADAppRoleAssignment -ClientId $MainAppClientId -Role $RoleName -RemoveRoleFromClientId $ClientAppClientId -RemoveRoleIfNoAssignmentsAreLeft
+                }
             }
             It "Removing a role assignment and role should succeed" {
                 # Arrange
-                $ClientId = $config.Arcus.ActiveDirectory.MainAppClientId
+                $MainAppClientId = $config.Arcus.ActiveDirectory.MainAppClientId
                 $RoleName = 'SomeRole'
-                $RemoveRoleFromClientId = $config.Arcus.ActiveDirectory.ClientAppClientId
+                $ClientAppClientId = $config.Arcus.ActiveDirectory.ClientAppClientId
+                Add-AzADAppRoleAssignment -ClientId $MainAppClientId -Role $RoleName -AssignRoleToClientId $ClientAppClientId
 
-                # Act
-                {
-                   Remove-AzADAppRoleAssignment -ClientId $ClientId -Role $RoleName -RemoveRoleFromClientId $RemoveRoleFromClientId -RemoveRoleIfNoAssignmentsAreLeft
-                } | Should -Not -Throw
+                try {
+                    # Act
+                    {
+                       Remove-AzADAppRoleAssignment -ClientId $MainAppClientId -Role $RoleName -RemoveRoleFromClientId $ClientAppClientId -RemoveRoleIfNoAssignmentsAreLeft
+                    } | Should -Not -Throw
 
-                # Assert
-                $adApplication = Get-AzADApplication -Filter "AppId eq '$ClientId'"
-                $RoleName | Should -Not -BeIn $adApplication.AppRole.Value
+                    # Assert
+                    $adApplication = Get-AzADApplication -Filter "AppId eq '$MainAppClientId'"
+                    $RoleName | Should -Not -BeIn $adApplication.AppRole.Value
 
-                $adServicePrincipal = Get-AzADServicePrincipal -Filter "AppId eq '$ClientId'"
-                $appRoleAssignments = Get-MgServicePrincipalAppRoleAssignedTo -ServicePrincipalId $adServicePrincipal.Id
-                $appRoleAssignments.Count | Should -Be 0
+                    $adServicePrincipal = Get-AzADServicePrincipal -Filter "AppId eq '$MainAppClientId'"
+                    $appRoleAssignments = Get-MgServicePrincipalAppRoleAssignedTo -ServicePrincipalId $adServicePrincipal.Id
+                    $appRoleAssignments.Count | Should -Be 0
+                } finally {
+                    Remove-AzADAppRoleAssignment -ClientId $MainAppClientId -Role $RoleName -RemoveRoleFromClientId $ClientAppClientId -RemoveRoleIfNoAssignmentsAreLeft
```

Comment:
This should then be add, no?

Reply:
It should be changed to `Add-` something, right?

Reply:
It should be remove since the original state of the AD is to not have the role and assignment. But I removed this try catch [here](https://github.com/arcus-azure/arcus.scripting/pull/336/commits/2c4d9d5180a95ad3832a440e659fa70bda810158) since it does not make a lot of sense looking at it now. The test already does a `Remove-AzADAppRoleAssignment` itself, it that somehow fails the finally will also fail since it does the same command.

Reply:
Ah, sorry, in this case, it should be a 'setup' clause where we add the role assignment.

Reply:
Yes, I am doing that in the Arrange step here https://github.com/arcus-azure/arcus.scripting/blob/ed7f050be4a534b661f10847c278d94fb1a3b700/src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ActiveDirectory.tests.ps1#L56